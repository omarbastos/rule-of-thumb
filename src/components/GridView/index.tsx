import CardSquare from 'components/base/Cards/Square'
import React from 'react'
import { Poll } from 'utils/interfaces'

interface Props {
  polls: Poll[]
}

const GridView = ({ polls }: Props) => {
  const SKELETONS = 16
  if (polls.length === 0) {
    return (
      <div className="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-4">
        {[...Array(SKELETONS)].map(() => (
          <div className="w-60 h-24 border-2 rounded-md mx-auto my-2">
            <div className="flex animate-pulse flex-row items-center h-full justify-center space-x-5">
              <div className="w-12 bg-gray-300 h-12 rounded-full "></div>
              <div className="flex flex-col space-y-3">
                <div className="w-36 bg-gray-300 h-6 rounded-md "></div>
                <div className="w-24 bg-gray-300 h-6 rounded-md "></div>
              </div>
            </div>
          </div>
        ))}
      </div>
    )
  }
  return (
    <div className="flex overflow-x-scroll no-scrollbar md:grid md:grid-cols-2 lg:grid-cols-3 md:gap-4 justify-items-center items-center md:justify-center">
      {polls.map((poll: Poll) => (
        <CardSquare key={poll.uid} {...poll} />
      ))}
    </div>
  )
}

export default GridView
