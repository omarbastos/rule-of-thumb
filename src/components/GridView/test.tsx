import { render, screen } from '@testing-library/react'

import GridView from '.'

describe('<App />', () => {
  it('should render App as default', () => {
    const { container } = render(<GridView polls={[]} />)

    expect(container).toMatchSnapshot()
  })
})
