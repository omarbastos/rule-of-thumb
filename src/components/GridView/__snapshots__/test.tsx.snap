// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render App as default 1`] = `
<div>
  <div
    class="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-4"
  >
    <div
      class="w-60 h-24 border-2 rounded-md mx-auto my-2"
    >
      <div
        class="flex animate-pulse flex-row items-center h-full justify-center space-x-5"
      >
        <div
          class="w-12 bg-gray-300 h-12 rounded-full "
        />
        <div
          class="flex flex-col space-y-3"
        >
          <div
            class="w-36 bg-gray-300 h-6 rounded-md "
          />
          <div
            class="w-24 bg-gray-300 h-6 rounded-md "
          />
        </div>
      </div>
    </div>
    <div
      class="w-60 h-24 border-2 rounded-md mx-auto my-2"
    >
      <div
        class="flex animate-pulse flex-row items-center h-full justify-center space-x-5"
      >
        <div
          class="w-12 bg-gray-300 h-12 rounded-full "
        />
        <div
          class="flex flex-col space-y-3"
        >
          <div
            class="w-36 bg-gray-300 h-6 rounded-md "
          />
          <div
            class="w-24 bg-gray-300 h-6 rounded-md "
          />
        </div>
      </div>
    </div>
    <div
      class="w-60 h-24 border-2 rounded-md mx-auto my-2"
    >
      <div
        class="flex animate-pulse flex-row items-center h-full justify-center space-x-5"
      >
        <div
          class="w-12 bg-gray-300 h-12 rounded-full "
        />
        <div
          class="flex flex-col space-y-3"
        >
          <div
            class="w-36 bg-gray-300 h-6 rounded-md "
          />
          <div
            class="w-24 bg-gray-300 h-6 rounded-md "
          />
        </div>
      </div>
    </div>
    <div
      class="w-60 h-24 border-2 rounded-md mx-auto my-2"
    >
      <div
        class="flex animate-pulse flex-row items-center h-full justify-center space-x-5"
      >
        <div
          class="w-12 bg-gray-300 h-12 rounded-full "
        />
        <div
          class="flex flex-col space-y-3"
        >
          <div
            class="w-36 bg-gray-300 h-6 rounded-md "
          />
          <div
            class="w-24 bg-gray-300 h-6 rounded-md "
          />
        </div>
      </div>
    </div>
    <div
      class="w-60 h-24 border-2 rounded-md mx-auto my-2"
    >
      <div
        class="flex animate-pulse flex-row items-center h-full justify-center space-x-5"
      >
        <div
          class="w-12 bg-gray-300 h-12 rounded-full "
        />
        <div
          class="flex flex-col space-y-3"
        >
          <div
            class="w-36 bg-gray-300 h-6 rounded-md "
          />
          <div
            class="w-24 bg-gray-300 h-6 rounded-md "
          />
        </div>
      </div>
    </div>
    <div
      class="w-60 h-24 border-2 rounded-md mx-auto my-2"
    >
      <div
        class="flex animate-pulse flex-row items-center h-full justify-center space-x-5"
      >
        <div
          class="w-12 bg-gray-300 h-12 rounded-full "
        />
        <div
          class="flex flex-col space-y-3"
        >
          <div
            class="w-36 bg-gray-300 h-6 rounded-md "
          />
          <div
            class="w-24 bg-gray-300 h-6 rounded-md "
          />
        </div>
      </div>
    </div>
    <div
      class="w-60 h-24 border-2 rounded-md mx-auto my-2"
    >
      <div
        class="flex animate-pulse flex-row items-center h-full justify-center space-x-5"
      >
        <div
          class="w-12 bg-gray-300 h-12 rounded-full "
        />
        <div
          class="flex flex-col space-y-3"
        >
          <div
            class="w-36 bg-gray-300 h-6 rounded-md "
          />
          <div
            class="w-24 bg-gray-300 h-6 rounded-md "
          />
        </div>
      </div>
    </div>
    <div
      class="w-60 h-24 border-2 rounded-md mx-auto my-2"
    >
      <div
        class="flex animate-pulse flex-row items-center h-full justify-center space-x-5"
      >
        <div
          class="w-12 bg-gray-300 h-12 rounded-full "
        />
        <div
          class="flex flex-col space-y-3"
        >
          <div
            class="w-36 bg-gray-300 h-6 rounded-md "
          />
          <div
            class="w-24 bg-gray-300 h-6 rounded-md "
          />
        </div>
      </div>
    </div>
    <div
      class="w-60 h-24 border-2 rounded-md mx-auto my-2"
    >
      <div
        class="flex animate-pulse flex-row items-center h-full justify-center space-x-5"
      >
        <div
          class="w-12 bg-gray-300 h-12 rounded-full "
        />
        <div
          class="flex flex-col space-y-3"
        >
          <div
            class="w-36 bg-gray-300 h-6 rounded-md "
          />
          <div
            class="w-24 bg-gray-300 h-6 rounded-md "
          />
        </div>
      </div>
    </div>
    <div
      class="w-60 h-24 border-2 rounded-md mx-auto my-2"
    >
      <div
        class="flex animate-pulse flex-row items-center h-full justify-center space-x-5"
      >
        <div
          class="w-12 bg-gray-300 h-12 rounded-full "
        />
        <div
          class="flex flex-col space-y-3"
        >
          <div
            class="w-36 bg-gray-300 h-6 rounded-md "
          />
          <div
            class="w-24 bg-gray-300 h-6 rounded-md "
          />
        </div>
      </div>
    </div>
    <div
      class="w-60 h-24 border-2 rounded-md mx-auto my-2"
    >
      <div
        class="flex animate-pulse flex-row items-center h-full justify-center space-x-5"
      >
        <div
          class="w-12 bg-gray-300 h-12 rounded-full "
        />
        <div
          class="flex flex-col space-y-3"
        >
          <div
            class="w-36 bg-gray-300 h-6 rounded-md "
          />
          <div
            class="w-24 bg-gray-300 h-6 rounded-md "
          />
        </div>
      </div>
    </div>
    <div
      class="w-60 h-24 border-2 rounded-md mx-auto my-2"
    >
      <div
        class="flex animate-pulse flex-row items-center h-full justify-center space-x-5"
      >
        <div
          class="w-12 bg-gray-300 h-12 rounded-full "
        />
        <div
          class="flex flex-col space-y-3"
        >
          <div
            class="w-36 bg-gray-300 h-6 rounded-md "
          />
          <div
            class="w-24 bg-gray-300 h-6 rounded-md "
          />
        </div>
      </div>
    </div>
    <div
      class="w-60 h-24 border-2 rounded-md mx-auto my-2"
    >
      <div
        class="flex animate-pulse flex-row items-center h-full justify-center space-x-5"
      >
        <div
          class="w-12 bg-gray-300 h-12 rounded-full "
        />
        <div
          class="flex flex-col space-y-3"
        >
          <div
            class="w-36 bg-gray-300 h-6 rounded-md "
          />
          <div
            class="w-24 bg-gray-300 h-6 rounded-md "
          />
        </div>
      </div>
    </div>
    <div
      class="w-60 h-24 border-2 rounded-md mx-auto my-2"
    >
      <div
        class="flex animate-pulse flex-row items-center h-full justify-center space-x-5"
      >
        <div
          class="w-12 bg-gray-300 h-12 rounded-full "
        />
        <div
          class="flex flex-col space-y-3"
        >
          <div
            class="w-36 bg-gray-300 h-6 rounded-md "
          />
          <div
            class="w-24 bg-gray-300 h-6 rounded-md "
          />
        </div>
      </div>
    </div>
    <div
      class="w-60 h-24 border-2 rounded-md mx-auto my-2"
    >
      <div
        class="flex animate-pulse flex-row items-center h-full justify-center space-x-5"
      >
        <div
          class="w-12 bg-gray-300 h-12 rounded-full "
        />
        <div
          class="flex flex-col space-y-3"
        >
          <div
            class="w-36 bg-gray-300 h-6 rounded-md "
          />
          <div
            class="w-24 bg-gray-300 h-6 rounded-md "
          />
        </div>
      </div>
    </div>
    <div
      class="w-60 h-24 border-2 rounded-md mx-auto my-2"
    >
      <div
        class="flex animate-pulse flex-row items-center h-full justify-center space-x-5"
      >
        <div
          class="w-12 bg-gray-300 h-12 rounded-full "
        />
        <div
          class="flex flex-col space-y-3"
        >
          <div
            class="w-36 bg-gray-300 h-6 rounded-md "
          />
          <div
            class="w-24 bg-gray-300 h-6 rounded-md "
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
